e2e/eval_priority/eval_priority.go

```diff
@@ -0,0 +1,214 @@
+package eval_priority
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/e2eutil"
+	"github.com/hashicorp/nomad/e2e/framework"
+	"github.com/hashicorp/nomad/helper/uuid"
+)
+
+type EvalPriorityTest struct {
+	framework.TC
+	jobIDs []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "EvalPriority",
+		CanRunLocal: true,
+		Cases: []framework.TestCase{
+			new(EvalPriorityTest),
+		},
+	})
+}
+
+func (tc *EvalPriorityTest) BeforeAll(f *framework.F) {
+	e2eutil.WaitForLeader(f.T(), tc.Nomad())
+	e2eutil.WaitForNodesReady(f.T(), tc.Nomad(), 1)
+}
+
+func (tc *EvalPriorityTest) AfterEach(f *framework.F) {
+	for _, id := range tc.jobIDs {
+		_, _, err := tc.Nomad().Jobs().Deregister(id, true, nil)
+		f.NoError(err)
+	}
+	tc.jobIDs = []string{}
+
+	_, err := e2eutil.Command("nomad", "system", "gc")
+	f.NoError(err)
+}
+
+// TestJobRegisterWithoutEvalPriority makes sure that when not specifying an
+// eval priority on job register, the job priority is used.
+func (tc *EvalPriorityTest) TestJobRegisterWithoutEvalPriority(f *framework.F) {
```

Comment:
I love that we're being thorough with E2E, but the priority set on the `Evaluation` object seems like it's already well-covered in the other tests you've written. If we're going to go through the "cost" of an E2E test, is it possible to exercise that the priorities are respected by the scheduler and deployment as we'd expect?

Reply:
Yeah I certainly got tunnel vision here. I'll rework the E2E as per your suggestion.